Tag and click arrangement

replaceafill opened the pull request:
This PR:

Connects to https://github.com/archivematica/Issues/issues/369

- Synchronizes tag display/add/removal in all the panes of the Appraisal tab
- Adds **tag-n-click** functionality to the Appraisal tab allowing selection of all the files with the same tag to be included in a SIP

Connects to https://github.com/archivematica/Issues/issues/507

- Merges the two Angular apps (appraisal-tab and transfer-browser) into a single dashboard app, synchronizing their dependencies and adds browser configurations to the dashboard.Dockerfile to allow running JS tests with Karma

Connects to https://github.com/archivematica/Issues/issues/472

- Fixes tag removal in the File list pane of the Appraisal tab

replaceafill committed:
Fix and run Karma tests of Angular apps in Chrome and Firefox

replaceafill committed:
Sync Angular apps dependency versions

replaceafill committed:
Move appraisal-tab contents up to frontend directory

replaceafill committed:
Rename appraisal-tab app to dashboard

replaceafill committed:
Merge transfer-browser app into the dashboard

replaceafill committed:
Fix tag removal in Appraisal tab (#472)

replaceafill committed:
Add remove tag buttons in Backlog pane

replaceafill committed:
Add remove tag buttons in Arrangment pane

replaceafill committed:
Add create SIP from tags functionality

replaceafill committed:
Update deprecated prepublish script to prepare

replaceafill committed:
Add instructions how to merge and updated translations

sevein commented on src/dashboard.Dockerfile:
```diff
@@ -19,13 +19,64 @@ RUN set -ex \
 		locales \
 		locales-all \
 		unar \
+		wget \
```
@replaceafill do you think it's worth to move the Chrome/FF installer snippets over to use curl so we don't need to install wget too?

sevein commented on src/dashboard.Dockerfile:
```diff
@@ -19,13 +19,64 @@ RUN set -ex \
+	&& echo "Using chromedriver version: "$CD_VERSION \
+	&& wget --no-verbose -O /tmp/chromedriver_linux64.zip https://chromedriver.storage.googleapis.com/$CD_VERSION/chromedriver_linux64.zip \
+	&& rm -rf /opt/selenium/chromedriver \
+	&& unzip /tmp/chromedriver_linux64.zip -d /opt/selenium \
```
I'm seeing this error:

    /bin/sh: 1: unzip: not found

It disappear when I had `unzip` installed. It'd also be possible to rewrite the snippet so it uses `unar` instead, which is already installed by us (see L21).

sevein reviewed (commented):
Neat! It works, including tests!

👏🎉👏🎉👏🎉👏🎉👏🎉 

I've added a couple of minor comments.

replaceafill committed:
Use curl instead of wget and unar instead of unzip for setting up the browsers

replaceafill commented:
@sevein I just addressed your observations.

I also fixed the `npm install` command. I probably didn't noticed before because I still had the old `frontend/appraisal-tab` and `frontend/transfer-browser` directories lying around.

sevein commented:
> I also fixed the npm install command. I probably didn't noticed before because I still had the old frontend/appraisal-tab and frontend/transfer-browser directories lying around.

I guess it happen the same to me, but I wasn't testing the new functionality you've added so I guess that's why I missed it! I'll give it another try tonight! Thanks you!

replaceafill committed:
Add travis setup for running dashboard JS tests

sevein commented on .travis.yml:
```diff
@@ -43,6 +43,26 @@ matrix:
+        - export DISPLAY=:99.0
+        - sh -e /etc/init.d/xvfb start
+        - sleep 3 # give xvfb some time to start
+        - cd ./src/dashboard/frontend/
```
Wow @replaceafill that was fast. Why do we still need xvfb?

replaceafill replied:
We don't. And this made me realize we don't need the browser drivers either in the `dashboard.Dockerfile`. I just removed them in https://github.com/artefactual/archivematica/pull/1364/commits/36bba916ca5930caa9f2aad13a070048f67dd988

sevein commented on .travis.yml:
```diff
@@ -43,6 +43,26 @@ matrix:
+      addons:
+        apt:
+          sources:
+            - google-chrome
```
Out of curiosity, why is this needed?

replaceafill replied:
I was just trying to mimic what we do in the `dashboard.Dockerfile` (download key, add source, etc). But I just simplified it in https://github.com/artefactual/archivematica/pull/1364/commits/36bba916ca5930caa9f2aad13a070048f67dd988 to:

```
addons:
  chrome: stable
```

replaceafill committed:
Remove unnecessary browser drivers and simplified travis job

replaceafill committed:
Update browser versions

sevein commented:
Thanks @replaceafill! 1 min 44 secs for build time is pretty great.

sevein reviewed (approved):
:heart_eyes:

ablwr commented:
So awesome, can't wait to check this out!
